main.go

```diff
@@ -1,9 +1,62 @@
 package main
 
 import (
-	"github.com/Sirupsen/logrus"
+	"fmt"
+	"net"
+	"os"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/rancher/rancher-cni-ipam/ipfinder/metadata"
 )
 
+func cmdAdd(args *skel.CmdArgs) error {
+	ipamConf, err := LoadIPAMConfig(args.StdinData, args.Args)
+	if err != nil {
+		return err
+	}
+
+	if ipamConf.LogToFile != "" {
+		f, err := os.OpenFile(ipamConf.LogToFile, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+		if err == nil && f != nil {
+			log.SetOutput(f)
+			defer f.Close()
+		}
+	}
+
+	log.Debugf("rancher-cni-ipam: cmdAdd: invoked")
+	log.Debugf("rancher-cni-ipam: %s", fmt.Sprintf("args: %#v", args))
+	log.Debugf("rancher-cni-ipam: %s", fmt.Sprintf("ipamConf: %#v", ipamConf))
+
+	ipf, err := metadata.NewIPFinderFromMetadata()
+	if err != nil {
+		return err
+	}
+	ipString := ipf.GetIP(args.ContainerID)
+
+	log.Debugf("rancher-cni-ipam: %s", fmt.Sprintf("ip: %#v", ipString))
+
+	ip, ipnet, err := net.ParseCIDR(ipString + "/16")
```

Comment:
Only the mask is configurable, right now we are getting the IP directly from cattle. There is no concept of separate subnet range for different hosts. I have the changes for the mask in the configuration. Will submit them in the next PR.